app/data/index.js

```diff
@@ -43,6 +43,10 @@ module.exports = Base => class extends Base {
         return new Promise((resolve, reject) => {
             Player.findOne({ name: playerName }, (err, player) => {
                 if (err) return reject(err);
+                
+                if (player.shareHomes && player.shareHomes.hasOwnProperty(homeName)) {
```

Comment:
This is an anti pattern and shouldn't be happening here. We may need to have a discussion about design pattern on this